cli/command/load_balancer.go

```diff
@@ -10,6 +17,447 @@ func NewLoadBalancerCommand(b *CommandBase) *LoadBalancerCommand {
+						Value: 2,
+						Usage: "Number of consecutive successes required to count as healthy",
+					},
+					cli.IntFlag{
+						Name:  "healthcheck-unhealthy-threshold",
+						Value: 2,
+						Usage: "Number of consecutive failures required to count as unhealthy",
+					},
+				},
+			},
+			{
+				Name:      "delete",
+				Usage:     "Delete the load balancer LOADBALANCER_NAME",
+				ArgsUsage: "LOADBALANCER_NAME",
+				Action:    l.delete,
+			},
+			{
+				Name:      "dropport",
+				Usage:     "Drop the listener with host port HOST_PORT from load balancer LOADBALANCER_NAME",
+				Action:    l.dropport,
+				ArgsUsage: "LOADBALANCER_NAME HOST_PORT",
+			},
+			{
+				Name:      "get",
+				Usage:     "Describe load balancer LOADBALANCER_NAME",
+				Action:    l.read,
+				ArgsUsage: "LOADBALANCER_NAME",
+			},
+			{
+				Name:      "healthcheck",
+				Usage:     "View or update the health check of load balancer LOADBALANCER_NAME. If no options are specified, the health check is displayed.",
+				Action:    l.healthcheck,
+				ArgsUsage: "LOADBALANCER_NAME",
+				Flags: []cli.Flag{
+					cli.StringFlag{
+						Name:  "healthcheck-target",
+						Usage: "Health check target in format 'PROTOCOL:PORT' or 'PROTOCOL:PORT/WITH/PATH'",
+					},
+					cli.IntFlag{
+						Name:  "healthcheck-interval",
+						Usage: "Health check interval in seconds",
+					},
+					cli.IntFlag{
+						Name:  "healthcheck-timeout",
+						Usage: "Health check timeout in seconds",
+					},
+					cli.IntFlag{
+						Name:  "healthy-threshold",
+						Usage: "Number of consecutive successes required to count as healthy",
+					},
+					cli.IntFlag{
+						Name:  "unhealthy-threshold",
+						Usage: "Number of consecutive failures required to count as unhealthy",
+					},
+				},
+			},
+			{
+				Name:      "list",
+				Usage:     "List all load balancers",
+				Action:    l.list,
+				ArgsUsage: " ",
+			},
+		},
+	}
+}
+
+func (l *LoadBalancerCommand) addport(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "LOADBALANCER_NAME", "PORT")
+	if err != nil {
+		return err
+	}
+
+	port, err := parsePort(args["PORT"], c.String("certificate"))
+	if err != nil {
+		return err
+	}
+
+	loadBalancerID, err := l.resolveSingleEntityIDHelper("load_balancer", args["LOADBALANCER_NAME"])
+	if err != nil {
+		return err
+	}
+
+	loadBalancer, err := l.client.ReadLoadBalancer(loadBalancerID)
+	if err != nil {
+		return err
+	}
+
+	ports := append(loadBalancer.Ports, *port)
+	req := models.UpdateLoadBalancerRequest{
+		LoadBalancerID: loadBalancerID,
+		Ports:          &ports,
+	}
+
+	jobID, err := l.client.UpdateLoadBalancer(req)
+	if err != nil {
+		return err
+	}
+
+	onCompleteFN := func(loadBalancerID string) error {
+		loadBalancer, err := l.client.ReadLoadBalancer(loadBalancerID)
+		if err != nil {
+			return err
+		}
+
+		l.printer.StopSpinner()
+		return l.printer.PrintLoadBalancers(loadBalancer)
+	}
+
+	return l.waitOnJobHelper(c, jobID, "adding port", onCompleteFN)
```

Comment:
[nit] personally find defining the function `onCompleteFN()` as an anonymous function within the call to `waitOnJobHelper()` to be more readable